### ISSUE TITLE
Mass action DHL Parcel does not work when using "Select all"

### ISSUE THREAD

**Issue by robinero** [dhlparcel/plugin-magento2-release#18]
Using: Magento 2.3.5-p1.

When in the Sales -> Orders admin grid, using the "Select all" option on a single order (after for example filtering) or a set of orders, "DHL Parcel: Create Labels" returns the following error:

"Something unexpected happened, please contact your administrator"

This is caused by `$this->_request->getParam('selected')` returning `NULL` on:
https://github.com/dhlparcel/plugin-magento2-release/blob/master/Controller/Adminhtml/Bulk/Create.php#L51

When manually selecting orders (eg. 5 out of 10 orders), the request param does return a set of IDs, as to why this happens I have no idea.

I suggest implementing `Magento\Ui\Component\MassAction\Filter`:

```
use Magento\Ui\Component\MassAction\Filter;
use Magento\Sales\Model\ResourceModel\Order\CollectionFactory;
```

Instead of:
```php
$this->_request->getParam('selected')
```
Create a collection to iterate over:
```php
$collection = $this->filter->getCollection($this->collectionFactory->create());
```

So the action will look something like this:
```php
$collection = $this->filter->getCollection($this->collectionFactory->create());

foreach ($collection as $order) {
    try {
        $this->orderService->createShipment($order);
        $success[] = '#' . $order->getRealOrderId();
        $successOrderIds[] = $order->getRealOrderId();
    } catch (LocalizedException $e) {
        $errors['#' . $order->getRealOrderId()] = $e;
    }
}
```

You can find a Magento example in:  
https://github.com/magento/magento2/blob/2.3/app/code/Magento/Customer/Controller/Adminhtml/Address/MassDelete.php#L88

**Comment by FedeltaMedia** [comment]
Maybe #17 is related? I also have issues with the bulk selection menu and the 'Create labels'. I do however get a different error though.

**Comment by roerlemans** [comment]
@robinero thanks for your ticket! We have this implemented in our next release. Thanks for your solution.

**Comment by roerlemans** [comment]
This bug is fixed in v1.0.16:
https://github.com/dhlparcel/plugin-magento2-release/releases/tag/v1.0.16
